### DISCUSSION TITLE
Lesson 3 NFT : Spun up local chain (anvil) to solve challenge, while doing cast send and failed

### DISCUSSION THREAD

**Discussion by BLTC-520** [Cyfrin/security-and-auditing-full-course-s23#253] [General]
My DeployS3.s.sol : 
```
// SPDX-License-Identifier: MIT
pragma solidity ^0.8.20;

import {Script} from "forge-std/Script.sol";
import {S3} from "../src/S3.sol";
import {console} from "forge-std/console.sol";

contract DeployS3 is Script {
    function run() external {
        // Load the deployer's private key
        uint256 deployerPrivateKey = vm.envUint("PRIVATE_KEY");

        // Start broadcasting transactions
        vm.startBroadcast(deployerPrivateKey);

        // Replace the registry address with the actual address you want to use
        address registry = 0x90F79bf6EB2c4f870365E785982E1f101E93b906;

        // Deploy the S3 contract
        S3 s3 = new S3(registry);

        console.log("S3 deployed to:", address(s3));

        // Stop broadcasting
        vm.stopBroadcast();
    }
}
```

0x057ef64E23666F000b34aE31332854aCBd1c8544 -> this is the address which deployed to. 

Then I use cast storage -> it indeed have 0x7b (123) inside. 

But when I try to do this command: 
cast send 0x057ef64E23666F000b34aE31332854aCBd1c8544 "solveChallenge(uint256,string)" 123 "" --rpc-url http://127.0.0.1:8545 --private-key $PRIVATE_KEY 

fyi, $PRIVATE_KEY already using source .env command to save and successfully echo. 

`Error: server returned an error response: error code 3: execution reverted, data: "0x"`

This error occurred. Anyone can help give some hint?

**Comment by EngrPips** [comment]
Hello @BLTC-520, Are you trying to solve the challenge at the end of section 3 of the course to get the course NFT?

**Reply by BLTC-520** [reply]
Yes.

**Reply by EngrPips** [reply]
Why are you solving it on the Anvil chain? The course. NFT, is not deployed to the Anvil chain. The image below shows the chain on which the course NFT is deployed.

![Screenshot 2024-11-27 at 17 41 02](https://github.com/user-attachments/assets/54d5341a-d599-4300-b1de-dbcc3c40e552)

**Reply by BLTC-520** [reply]
lol u point out a very good point, me dumbass..... 😂 So, the rpc-url set to Sepolia will do right?

**Reply by EngrPips** [reply]
Please read this discussion https://github.com/Cyfrin/foundry-full-course-cu/discussions/2018. It should guide you on how to solve the NFT challenge.

**Reply by BLTC-520** [reply]
Thanks mate!

**Reply by EngrPips** [reply]
Anytime, Friend.
